src/core/ext/filters/client_channel/lb_policy/round_robin/round_robin.c

```diff
@@ -421,16 +425,15 @@ static grpc_connectivity_state update_lb_connectivity_status_locked(
                                 GRPC_CHANNEL_CONNECTING, GRPC_ERROR_NONE,
                                 "rr_connecting");
     return GRPC_CHANNEL_CONNECTING;
-  } else if (p->num_subchannels == 0) { /* 3) SHUTDOWN */
+  } else if (p->num_shutdown == p->num_subchannels) { /* 3) SHUTDOWN */
     grpc_connectivity_state_set(exec_ctx, &p->state_tracker,
-                                GRPC_CHANNEL_SHUTDOWN, GRPC_ERROR_REF(error),
```

Comment:
I see.

This fixes the problem if we happen to hit the branches for cases 3 or 4, but it leaves the leak in case we hit any of the other branches.  While it is probably true that error will be GRPC_ERROR_NONE in any of the other cases, it seems safer to not depend on that assumption.

I think the right fix here is to leave the GRPC_ERROR_REF in place in these two branches, but add a GRPC_ERROR_UNREF right before we return from this function.